contracts/factories/ManagerFactory.sol

```diff
@@ -0,0 +1,201 @@
+/*
+    Copyright 2021 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental ABIEncoderV2;
+
+import { ISetToken } from "@setprotocol/set-protocol-v2/contracts/interfaces/ISetToken.sol";
+
+import { DelegatedManager } from "../manager/DelegatedManager.sol";
+import { IDelegatedManager } from "../interfaces/IDelegatedManager.sol";
+import { ISetTokenCreator } from "../interfaces/ISetTokenCreator.sol";
+
+contract ManagerFactory {
+    struct InitializeParams{
+        address deployer;
+        address owner;
+        IDelegatedManager manager;
+        bool isPending;
+    }
+    
+    ISetTokenCreator public factory;
+    mapping(ISetToken=>InitializeParams) public initializeState;
+    mapping(ISetToken=>bool) public isValidSet;
+    address[] internal validSets; 
+
+    constructor(
+        ISetTokenCreator _setTokenFactory
+    ) public {
+        factory = _setTokenFactory;
+    }
+
+    /* ============ External Functions ============ */
+
+    function createSetAndManager(
+        address[] memory _components,
+        int256[] memory _units,
+        string memory _name,
+        string memory _symbol,
+        address _owner,
+        address _methodologist,
+        address[] memory _modules,
+        address[] memory _operators,
+        address[] memory _assets,
+        address[] memory _extensions
+    )
+        external
+        returns (ISetToken, address)
+    {
+        require(_extensions.length > 0, "Must have at least 1 extension");
+        // Require that components of Set are contained in the _assets array
+        // Require there be at least one operator?
+
+        ISetToken setToken = _deploySet(
+            _components,
+            _units,
+            _modules,
+            _name,
+            _symbol
+        );
+
+        DelegatedManager manager = _deployManager(
+            setToken,
+            _methodologist,
+            _extensions,
+            _operators,
+            _assets
+        );
+
+        _setInitializationState(setToken, address(manager), _owner);
+
```

Comment:
Event:

##### DelegatedManagerDeployed

| Type  | Name  | Description   |
|------ |------ |-------------  |
|address|setToken|Address of SetToken to be managed|
|address|deployer|Address of the deployer|
|address|owner|Address of the owner|
|address|manager|Address of the DelegatedManager|
|address[]|operators|Addresses of the operators|
|address[]|assets|Addresses of whitelisted assets|
|address[]|extensions|Addresses of extensions for manager|